.github/workflows/test_plugin.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  schedule:
+    # runs once a week
+    # - cron: '0 6 * * 0'
+    # runs every day
+    - cron: '0 12 * * *'
+  push:
+    branches: [ "main"]
+  pull_request:
+    branches: [ "master" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Pull qgis image
+      run: docker pull qgis/qgis:stable
+
+    - name: Run tests
+      run: docker run --net=host --volume $(pwd):/app -w=/app qgis/qgis:stable sh -c "python3 -m pip install pytest-qgis --break-system-packages && xvfb-run -s '+extension GLX -screen 0 1024x768x24' python3 -m pytest tests -s"
```

Comment:
What is the net for?

Reply:
To be honest I'm not sure. I just got it to run, but it would be nice to e.g. separate it into several tasks (like installing the pytest-qgis requirement in one and running the test in another) and also testing on Mac and Windows. But I'm not sure how to set that up